module/VuFind/src/VuFind/ServiceManager/ServiceInitializer.php

```diff
@@ -92,6 +92,11 @@ public function __invoke(ContainerInterface $sm, $instance)
         if ($instance instanceof \VuFind\I18n\Translator\TranslatorAwareInterface) {
             $instance->setTranslator($sm->get(\Laminas\Mvc\I18n\Translator::class));
         }
+        if ($instance instanceof \VuFind\Http\CachingDownloaderAwareInterface) {
```

Comment:
Is it worth adding "magic injection" support for this? I think this is a pattern Laminas is trying to move away from, so I'm not sure if we want to continue to embrace it with new functionality. (If you think it's important/useful, I won't object further... but in general I'd prefer to do more with factories and less with initializers). That's not to say that I wouldn't keep the associated Interface and Trait, as those are useful independently of the initializer.